pkg/appconsts/appconsts.go

```diff
@@ -3,10 +3,72 @@ package appconsts
 import (
 	"bytes"
 
-	"github.com/tendermint/tendermint/pkg/consts"
+	"github.com/celestiaorg/nmt/namespace"
+	"github.com/celestiaorg/rsmt2d"
 )
 
-// MaxShareVersion is the maximum value a share version can be.
-const MaxShareVersion = 127
+// These constants are sourced from:
+// https://github.com/celestiaorg/celestia-specs/blob/master/src/specs/consensus.md#constants
+const (
+	// ShareSize is the size of a share (in bytes).
+	// See: https://github.com/celestiaorg/celestia-specs/blob/master/src/specs/consensus.md#constants
+	ShareSize = 256
 
-var NameSpacedPaddedShareBytes = bytes.Repeat([]byte{0}, consts.MsgShareSize)
+	// NamespaceSize is the namespace size in bytes.
+	NamespaceSize = 8
+
+	// ShareReservedBytes is the reserved bytes for contiguous appends.
+	ShareReservedBytes = 1
+
+	// TxShareSize is the number of bytes usable for tx/evidence/ISR shares.
+	TxShareSize = ShareSize - NamespaceSize - ShareReservedBytes
+	// MsgShareSize is the number of bytes usable for message shares.
+	MsgShareSize = ShareSize - NamespaceSize
+
+	// MaxSquareSize is the maximum number of
+	// rows/columns of the original data shares in square layout.
+	// Corresponds to AVAILABLE_DATA_ORIGINAL_SQUARE_MAX in the spec.
+	// 128*128*256 = 4 Megabytes
+	// TODO(ismail): settle on a proper max square
+	// if the square size is larger than this, the block producer will panic
+	MaxSquareSize = 128
+	// MaxShareCount is the maximum number of shares allowed in the original data square.
+	// if there are more shares than this, the block producer will panic.
+	MaxShareCount = MaxSquareSize * MaxSquareSize
+
+	// MinSquareSize depicts the smallest original square width. A square size smaller than this will
+	// cause block producer to panic
+	MinSquareSize = 1
+	// MinshareCount is the minimum shares required in an original data square.
+	MinShareCount = MinSquareSize * MinSquareSize
+
+	// MaxShareVersion is the maximum value a share version can be.
+	MaxShareVersion = 127
+)
+
+var (
+	// IntermediateStateRootsNamespaceID is the namespace reserved for
+	// intermediate state root data
+	// TODO(liamsi): code commented out but kept intentionally.
+	// IntermediateStateRootsNamespaceID = namespace.ID{0, 0, 0, 0, 0, 0, 0, 2}
+
```

Comment:
[nit][blocking question]
wdyt about forwarding a link here to the missing constants? This way people don't have to remember which ones are where

like 
```go
TxNamespaceID = consts.TxNamespaceID
```

Reply:
if not, I'm also ok with approving as is. I think we're the only ones that use them

Reply:
Great idea, will do for all consts still in celestia-core

Reply:
Resolved in https://github.com/celestiaorg/celestia-app/pull/709/commits/6dde9c0f9d6ae56eda9eed6f5c28b3a1cb44e586